### ISSUE TITLE
'outline' property exposes implementation differences

### ISSUE THREAD

**Issue by zcorpan** [w3c/webvtt#235]
https://zcorpan.github.io/live-webvtt-viewer/#style=%3A%3Acue+%7B%0Aoutline%3A500px+solid+red%0A%7D

We should consider whether we want 'outline' to be in the list of supported properties. What's the use case?

**Comment by zcorpan** [comment]
cc @eric-carlson @foolip @rickeyre

**Comment by foolip** [comment]
On which browsers does this give different results? Screenshots would be nice :)

**Comment by zcorpan** [comment]
Opera (Presto):
<img width="1031" alt="The outline bleeds outside of the video." src="https://cloud.githubusercontent.com/assets/244772/10993533/ca9eaaca-846f-11e5-83d9-56bf3cd75ebb.png">

Opera (Blink):
<img width="1032" alt="The outline covers the full width of the video, but is clipped at the cue's edges vertically." src="https://cloud.githubusercontent.com/assets/244772/10993540/d82f9d02-846f-11e5-9390-9551fd4391f3.png">

Safari:
<img width="1031" alt="The outline covers almost the full width of the video, and is clipped a bit outside the cue's edges vertically." src="https://cloud.githubusercontent.com/assets/244772/10993545/e31782fc-846f-11e5-99e8-d137d055166b.png">

**Comment by foolip** [comment]
@eric-carlson @jernoble, is WebKit adding some padding to the cue background box, or something?

**Comment by zcorpan** [comment]
Gecko doesn't support styling yet AFAICT.

I suppose we can ignore Presto. The difference between Safari and Blink can be explained by Safari's default style not matching the spec -- it has a block background with rounded corners and padding, instead of an inline background box with `background:rgba(0,0,0,0.8)` as the spec says.

<img width="1031" alt="safari-no-style" src="https://cloud.githubusercontent.com/assets/244772/10993697/fca934c6-8470-11e5-8dc9-dd469043f312.png">

Still, I'm not certain that the spec's rendering rules, if implemented literally, would result in any of the behaviors we see here. I would have expected the outline to be clipped at the video element's edges, since that's the viewport for WebVTT.

**Comment by zcorpan** [comment]
Compare with http://software.hixie.ch/utilities/js/live-dom-viewer/saved/3737 (outline in an iframe)

**Comment by foolip** [comment]
Yeah, that is pretty odd. Doesn't border have any of these problems?

**Comment by zcorpan** [comment]
'border' is not in the list of supported properties (it affects layout).

**Comment by foolip** [comment]
Oh, right. How about background, then? It seems like the root cause here is deliberately different styles on the background box, so we should get that aligned somehow.

**Comment by zcorpan** [comment]
Yes, though it seems I can't select the root element in Safari to change the background; only `::cue(c)` etc works to select children. But yeah, we should decide between block and inline for the background box, and if it should be possible to switch to the other with CSS without having to use a `<c>` in every cue to get an inline background...

**Comment by silviapfeiffer** [comment]
The 5px gap in Safari likely comes from the "edge margin" which we introduced to cope with overscan. So, I think what Safari does is correct with the background - just not with the cue itself. ;-)

**Comment by zcorpan** [comment]
Neither of them is correct -- the video viewport is not that narrow vertically. The outline should fill the video viewport per spec.

**Comment by silviapfeiffer** [comment]
BTW: the use case is compatibility with CEA-708

**Comment by eric-carlson** [comment]
Apple's ports of WebKit take the system level caption user preferences into account (on OS X, System Preferences -> Accessibility -> Captions).

**Comment by foolip** [comment]
As of recently there is some of this in Blink as well, integrating with Android:
https://code.google.com/p/chromium/issues/detail?id=457850
https://developer.android.com/about/versions/kitkat.html#44-accessibility

However, it doesn't look like Android has any settings for window padding or rounded border radius, is that what iOS has that's at play here?

One problem is that the spec doesn't have any notion of a bounding box of all the text boxes of a cue that could be styled in the absence of regions. What does that box correspond to in WebKit? I can't find anything that seems to fiddle with padding or border radius, but it must be there somewhere :)

**Comment by silviapfeiffer** [comment]
The "cue box" is supposed to be the bounding box of all the text boxes.

**Comment by zcorpan** [comment]
Hmm, I noticed that the spec is probably wrong in that 'outline' applies to the cue box rather than the background box. Presto outlines the cue box (notice there is white on the sides of the text) but WebKit and Blink outline the background box.

**Comment by silviapfeiffer** [comment]
CEA708 puts the outline on the bounding box of all the text boxes, not on a full right-to-left covering box. Since it's implementing the same use case, 'cue box' seems more correct that 'background box' to me.

**Comment by foolip** [comment]
But WebVTT doesn't have a box for the bounding box of all the text boxes, the cue box is whatever size is specified by the `size` setting, subject to clamping.

**Comment by foolip** [comment]
I suppose that as part of https://github.com/w3c/webvtt/issues/249 we could add such a box, though.

**Comment by silviapfeiffer** [comment]
That's what I was hoping for. :-)
